client/components/payment-activity/test/index.test.tsx

```diff
@@ -163,4 +166,103 @@ describe( 'PaymentActivity component', () => {
 			queryByText( 'Are these metrics helpful?' )
 		).not.toBeInTheDocument();
 	} );
+
+	describe( 'Date selector renders correct ranges', () => {
+		const mockDateNowTo = ( date: string ) => {
+			Date.now = jest.fn( () =>
+				moment.tz( new Date( date ).getTime(), 'UTC' ).valueOf()
+			);
+		};
+		const dataSet = [
+			{
+				// Ordinary case or Happy Path
+				dateNow: '2024-06-10T16:19:29',
+				expected: {
+					today: 'June 10, 2024',
+					last7Days: 'June 3 - June 9, 2024',
+					last4Weeks: 'May 13 - June 9, 2024',
+					last3Months: 'March 10 - June 9, 2024',
+					last12Months: 'June 10, 2023 - June 9, 2024',
+					monthToDate: 'June 1 - June 10, 2024',
+					quarterToDate: 'April 1 - June 10, 2024',
+					yearToDate: 'January 1 - June 10, 2024',
+					allTime: 'January 1, 2022 - June 10, 2024',
+				},
+			},
+			{
+				// Start of the year
+				dateNow: '2024-01-01T00:00:00',
+				expected: {
+					today: 'January 1, 2024',
+					last7Days: 'December 25 - December 31, 2023',
+					last4Weeks: 'December 4 - December 31, 2023',
+					last3Months: 'October 1 - December 31, 2023',
+					last12Months: 'January 1 - December 31, 2023',
+					monthToDate: 'January 1, 2024',
+					quarterToDate: 'January 1, 2024',
+					yearToDate: 'January 1, 2024',
+					allTime: 'January 1, 2022 - January 1, 2024',
+				},
+			},
+			{
+				// Leap year
+				dateNow: '2024-02-29T00:00:00',
+				expected: {
+					today: 'February 29, 2024',
+					last7Days: 'February 22 - February 28, 2024',
+					last4Weeks: 'February 1 - February 28, 2024',
+					last3Months: 'November 29, 2023 - February 28, 2024',
+					last12Months: 'February 28, 2023 - February 28, 2024',
+					monthToDate: 'February 1 - February 29, 2024',
+					quarterToDate: 'January 1 - February 29, 2024',
+					yearToDate: 'January 1 - February 29, 2024',
+					allTime: 'January 1, 2022 - February 29, 2024',
+				},
+			},
+		];
+
+		it.each( dataSet )(
+			'should render the correct date ranges',
+			( { dateNow, expected } ) => {
+				mockDateNowTo( dateNow );
+
+				const { getAllByRole } = render( <PaymentActivity /> );
+
+				const dateSelectorButton = getAllByRole( 'button' )[ 0 ];
+				fireEvent.click( dateSelectorButton );
+				const datePresetOptions = getAllByRole( 'option' );
+
+				expect( datePresetOptions ).toHaveLength( 9 );
+				expect( datePresetOptions[ 0 ] ).toHaveTextContent(
+					`Today${ expected.today }`
+				);
```

Comment:
Tiny nitpick or discussion point: We can use `getByRole` here too, which is slightly more robust to the order of the options. (Unless we want to ensure this order strictly!)

We get to put the expected string with the space character too 🤏.

```suggestion
				getByRole( 'option', {
					name: `Today ${ expected.today }`,
				} );
```

Reply:
Superb suggestion! I have implemented it here. It looks cleaner without the array references and with the space character (it was making me a bit uneasy TBH 😃  -[bbc854b](https://github.com/Automattic/woocommerce-payments/pull/8961/commits/bbc854bd7155aeda91e33b4b37386a72e3297d40)